CONTRIBUTING.md

```diff
@@ -0,0 +1,213 @@
+# Contributing to build-definitions
+
+## Table of Contents
+
+* [Project Overview](#project-overview)
+* [How to Report Issues](#how-to-report-issues)
+  * [Bug Reports](#bug-reports)
+  * [Feature Requests](#feature-requests)
+* [How to Submit Pull Requests](#how-to-submit-pull-requests)
+  * [Development Workflow](#development-workflow)
+  * [Pull Request Guidelines](#pull-request-guidelines)
+  * [Code Quality Standards](#code-quality-standards)
+  * [Security Best Practices](#security-best-practices)
+* [Contribution Types](#contribution-types)
+  * [Adding New Tasks](#adding-new-tasks)
+  * [Modifying Existing Tasks](#modifying-existing-tasks)
+  * [Adding New Pipelines](#adding-new-pipelines)
+  * [Adding External Tasks](#adding-external-tasks)
+  * [Task Deprecation](#task-deprecation)
+* [Review Process](#review-process)
+
+## Project Overview
+
+This repository serves as a central repository for several components used in [Konflux](https://konflux-ci.dev).
+Therefore, everybody should follow its [Code of Conduct](https://github.com/konflux-ci/community/blob/main/CODE_OF_CONDUCT.md).
+
+This central repository contains components managed by the Konflux Build Team and other various contributors.
+See the CODEOWNERS file for more information.
+
+Components are delivered to Konflux via the `konflux-ci/tekton-catalog` Quay organization as OCI bundles.
+
+Note that this repository will be undergoing two major changes:
+1. Decentralization of tasks
+2. Replacement of Build team's task scripts with a new Go CLI
+
+## How to Report Issues
+
+We encourage early communication for all types of contributions. Before starting work, please open an issue to discuss your plans and get guidance from maintainers.
+
+### Bug Reports
+
+When reporting bugs, please include the following information:
+
+**Required Information:**
```

Comment:
nitpick: the **Required Information** header seems unnecessary, since there is no optional information